Fix issue with parallel downloads to a directory

qubixes opened the pull request:
See the linked issue for more details on what this PR fixes. I don't know if there are any side effects.

Fixes #543

trel commented:
looks good by inspection.

korydraughn commented:
Thanks for the contribution.

We need a test for this. You can take a swing at it or we can write one for you.

Do you have a preference?

qubixes commented:
@trel @korydraughn Thanks!

I have modified the `test_obj_put_get` test and split it into two test: one for small and one for large (as determined by the switch to parallel transfers). Does that work for you?

I haven't tested it locally, since I don't have a local iRODS server setup. If it's necessary I can try to create one at some point, but then it might be a bit longer.

korydraughn commented:
> Does that work for you?

That's fine.

Given you don't have a local server for testing, we will write a test for your PR.

You'll still get credit too. Thanks.

qubixes commented:
@korydraughn Thanks! I think the test I wrote should fail without the fix (and pass now), but perhaps you had something else in mind? Either way, I will leave it up to you then.

trel commented on irods/test/data_obj_test.py:
```diff
@@ -1317,7 +1323,7 @@ def test_obj_put_get(self):
 
         # make random 16M binary file
```
update comment

qubixes replied:
I have updated the comment and added two more comments for the tests.

trel replied:
perfect - thanks!

alanking reviewed (commented):
It looks good to me.

We can try running the test and if things work out we can use what you've written. Once confirmed, we can squash the commits to one or two and add an issue number to the commit message. Please hold on that until the test is confirmed working.

Thanks!

alanking reviewed (approved):
@d-w-moore has confirmed the tests are working as expected. Merging shortly. Thanks!!
